### ISSUE TITLE
[BUG] lemon package (from sqlite) replaces itself

### ISSUE THREAD

**Issue by Naine** [msys2/MSYS2-packages#2141]
## Describe the bug

Since the update to sqlite 3.33.0, the lemon package now wants to reinstall itself on every upgrade.

This line appears to be the cause: https://github.com/msys2/MSYS2-packages/blob/60f7466782967a9e4d643d187dda3cc1febb2640/sqlite/PKGBUILD#L230

## Steps to Reproduce the Problem

1. Perform upgrade by running `pacman -Syuu`. If core system upgrade occurred, follow with `pacman -Suu` to get full system upgrade, installing the version 3.33.0-1 sqlite and lemon packages. Skip if MSYS2 is up-to-date.
2. Install the lemon package if not already installed `pacman -S lemon`.
3. Repeat `pacman -Suu`. Observe the lemon package will want to replace itself with the same package and version.

Expected behavior: No new updates at step 3, and no query to `Replace lemon with msys/lemon?`

## Additional Context: Operating System, Screenshots

Console session below demonstrates:

```
$ pacman -Suu
:: Starting core system upgrade...
 there is nothing to do
:: Starting full system upgrade...
:: Replace lemon with msys/lemon? [Y/n] Y
:: Replace sqlite-compress with msys/sqlite-extensions? [Y/n] Y
:: Replace sqlite-icu with msys/sqlite? [Y/n] Y
resolving dependencies...
looking for conflicting packages...

Packages (16) binutils-2.35-1  lemon-3.33.0-1  lemon-3.32.3-1 [removal]  libsqlite-3.33.0-1  libsqlite-devel-3.33.0-1
              libxxhash-0.8.0-1  mingw-w64-cross-binutils-2.35-1  mingw-w64-i686-binutils-2.35-4
              mingw-w64-x86_64-binutils-2.35-4  mingw-w64-x86_64-harfbuzz-2.7.2-1  rsync-3.2.2-2  sqlite-3.33.0-1
              sqlite-compress-3.32.3-1 [removal]  sqlite-extensions-3.33.0-1  sqlite-icu-3.32.3-1 [removal]  tcl-8.6.10-1

Total Download Size:    35.02 MiB
Total Installed Size:  312.16 MiB
Net Upgrade Size:       13.99 MiB

:: Proceed with installation? [Y/n] Y
:: Retrieving packages...
 mingw-w64-i686-binutils-2.35-4-any              9.0 MiB  1254 KiB/s 00:07 [#########################################] 100%
 mingw-w64-x86_64-binutils-2.35-4-any            9.6 MiB  2.77 MiB/s 00:03 [#########################################] 100%
 mingw-w64-x86_64-harfbuzz-2.7.2-1-any        1038.3 KiB  2.23 MiB/s 00:00 [#########################################] 100%
 binutils-2.35-1-x86_64                          5.0 MiB  2.39 MiB/s 00:02 [#########################################] 100%
 tcl-8.6.10-1-x86_64                             2.4 MiB  2.62 MiB/s 00:01 [#########################################] 100%
 lemon-3.33.0-1-x86_64                          47.7 KiB  15.5 MiB/s 00:00 [#########################################] 100%
 libsqlite-3.33.0-1-x86_64                     552.2 KiB  2.21 MiB/s 00:00 [#########################################] 100%
 libsqlite-devel-3.33.0-1-x86_64               720.8 KiB  2.91 MiB/s 00:00 [#########################################] 100%
 libxxhash-0.8.0-1-x86_64                       22.4 KiB  21.9 MiB/s 00:00 [#########################################] 100%
 mingw-w64-cross-binutils-2.35-1-x86_64          5.1 MiB  2.51 MiB/s 00:02 [#########################################] 100%
 rsync-3.2.2-2-x86_64                          333.0 KiB  20.3 MiB/s 00:00 [#########################################] 100%
 sqlite-3.33.0-1-x86_64                       1000.9 KiB  3.85 MiB/s 00:00 [#########################################] 100%
 sqlite-extensions-3.33.0-1-x86_64             284.3 KiB  18.5 MiB/s 00:00 [#########################################] 100%
(13/13) checking keys in keyring                                           [#########################################] 100%
(13/13) checking package integrity                                         [#########################################] 100%
(13/13) loading package files                                              [#########################################] 100%
(13/13) checking for file conflicts                                        [#########################################] 100%
(16/16) checking available disk space                                      [#########################################] 100%
:: Processing package changes...
(1/3) removing sqlite-compress                                             [#########################################] 100%
(2/3) removing sqlite-icu                                                  [#########################################] 100%
(3/3) removing lemon                                                       [#########################################] 100%
( 1/13) upgrading binutils                                                 [#########################################] 100%
( 2/13) installing tcl                                                     [#########################################] 100%
( 3/13) installing lemon                                                   [#########################################] 100%
( 4/13) upgrading libsqlite                                                [#########################################] 100%
( 5/13) upgrading libsqlite-devel                                          [#########################################] 100%
( 6/13) upgrading libxxhash                                                [#########################################] 100%
( 7/13) upgrading mingw-w64-cross-binutils                                 [#########################################] 100%
( 8/13) upgrading mingw-w64-i686-binutils                                  [#########################################] 100%
( 9/13) upgrading mingw-w64-x86_64-binutils                                [#########################################] 100%
(10/13) upgrading mingw-w64-x86_64-harfbuzz                                [#########################################] 100%
(11/13) upgrading rsync                                                    [#########################################] 100%
(12/13) upgrading sqlite                                                   [#########################################] 100%
New optional dependencies for sqlite
    sqlite-extensions: various extensions for the SQL database engine 3.33.0 [installed]
(13/13) upgrading sqlite-extensions                                        [#########################################] 100%
:: Running post-transaction hooks...
(1/1) Updating the info directory file...

$ pacman -Suu
:: Starting core system upgrade...
 there is nothing to do
:: Starting full system upgrade...
:: Replace lemon with msys/lemon? [Y/n] Y
resolving dependencies...
looking for conflicting packages...

Packages (2) lemon-3.33.0-1  lemon-3.33.0-1 [removal]

Total Installed Size:   0.11 MiB
Net Upgrade Size:      -0.11 MiB

:: Proceed with installation? [Y/n] Y
(1/1) checking keys in keyring                                             [#########################################] 100%
(1/1) checking package integrity                                           [#########################################] 100%
(1/1) loading package files                                                [#########################################] 100%
(1/1) checking for file conflicts                                          [#########################################] 100%
(2/2) checking available disk space                                        [#########################################] 100%
:: Processing package changes...
(1/1) removing lemon                                                       [#########################################] 100%
(1/1) installing lemon                                                     [#########################################] 100%

$ pacman -Suu
:: Starting core system upgrade...
 there is nothing to do
:: Starting full system upgrade...
:: Replace lemon with msys/lemon? [Y/n] Y
resolving dependencies...
looking for conflicting packages...

Packages (2) lemon-3.33.0-1  lemon-3.33.0-1 [removal]

Total Installed Size:   0.11 MiB
Net Upgrade Size:      -0.11 MiB

:: Proceed with installation? [Y/n] n

```

**Comment by jeremyd2019** [comment]
Fixed by #2139

**Comment by angelog0** [comment]
Confirmed! It is in loop:

```
$ LANG=C pacman -Syu
:: Synchronizing package databases...
 mingw32 is up to date
 mingw64 is up to date
 msys is up to date
:: Starting core system upgrade...
 there is nothing to do
:: Starting full system upgrade...
:: Replace lemon with msys/lemon? [Y/n] Y
resolving dependencies...
looking for conflicting packages...

Packages (2) lemon-3.33.0-1  lemon-3.33.0-1 [removal]

Total Installed Size:   0.11 MiB
Net Upgrade Size:      -0.11 MiB

:: Proceed with installation? [Y/n] Y
(1/1) checking keys in keyring                                           [#######################################] 100%
(1/1) checking package integrity                                         [#######################################] 100%
(1/1) loading package files                                              [#######################################] 100%
(1/1) checking for file conflicts                                        [#######################################] 100%
(2/2) checking available disk space                                      [#######################################] 100%
:: Processing package changes...
(1/1) removing lemon                                                     [#######################################] 100%
(1/1) installing lemon                                                   [#######################################] 100%

$ LANG=C pacman -Syu
:: Synchronizing package databases...
 mingw32 is up to date
 mingw64 is up to date
 msys is up to date
:: Starting core system upgrade...
 there is nothing to do
:: Starting full system upgrade...
:: Replace lemon with msys/lemon? [Y/n]
```

How to fix this?

**Comment by Naine** [comment]
@angelog0 Only fix is to wait for 3.33.0-2 to be available, and just ignore until then.

Since #2139 has just been merged I'll close this issue. Thanks!
